60 script to tune hyperparameter and analyze

farrandi opened the pull request:
NOT DONE

atabak-alishiri committed:
draft of the scrip is added

atabak-alishiri committed:
python script for downloading and spliting data added

rbouwer committed:
fixede show_feat_coeff function and test, along with adapting FunctionTransformer parameter to feature_names_out='one-to-one'

farrandi committed:
update environment.yml to include click

farrandi committed:
update environment.yml to include click

farrandi committed:
initial draft of script

farrandi committed:
fixed environment.yml

farrandi committed:
fixed environment.yml

farrandi committed:
debugging preprocessor.get_feature_names_out() not working

farrandi committed:
add click to environment.yml

salva-u committed:
added the suggested folders within src

salva-u committed:
python script can ouput figures

salva-u committed:
python script can ouput figures

salva-u committed:
added pickle the preprocessor

salva-u committed:
added pickle the preprocessor

salva-u committed:
tester code you can use for unpickling model

salva-u committed:
tester code you can use for unpickling model

atabak-alishiri committed:
3 references added

atabak-alishiri committed:
all references added

atabak-alishiri committed:
Relocate file to notebook folder

atabak-alishiri committed:
the script now saves the sample data into a .zip file

farrandi committed:
add feature names out to fix preprocessor problem

atabak-alishiri committed:
Delete data/raw/test_dataset.csv

atabak-alishiri committed:
Delete data/raw/sampled_dataset.csv

atabak-alishiri committed:
Delete data/raw/train_dataset.csv

rbouwer commented on src/english_score_tuning.py:
```diff
@@ -0,0 +1,339 @@
+        scoring="RMSE",
+    )
+
+    # Discussion Section
```
I will split from here onwards into new script - I will make a PR tonight

farrandi committed:
update env

farrandi committed:
Add take test and train from file + output top model res to file

farrandi committed:
add one-to-one for feat_trf_out

farrandi committed:
add one-to-one for feat_trf_out

farrandi committed:
Get preprocessor from pickle

farrandi committed:
add jupyter_contrib_nbextensions=0.7.0 to env

farrandi committed:
fix verbose typos

farrandi commented:
- Added new input arguments to get train file, test file, preprocessor file, and set output dir (+ add defaults)
- Added and simplified the preprocessor pickle from #64 
- Added `jupyter_contrib_nbextenstions` to environment as per #71

salva-u committed:
added model/tables/ dir

salva-u committed:
rebased main

rbouwer commented on src/english_score_tuning.py:
```diff
@@ -0,0 +1,204 @@
+    "--preprocessor_path",
+    default="results/models/preprocessor/preprocessor.pkl",
+    help="Path to the preprocessor object (default: results/models/preprocessor/preprocessor.pkl)",
+)
```
Can we add a click option to store the fit preprocessor (so that I can access it in my new script):

@click.option(
    "--fit-pickle-to",
    type=str,
    help="Path to where the preprocessor (fitted to the training data) should be written ex: results/models/fit_preprocessor",
)

rbouwer commented on src/english_score_tuning.py:
```diff
@@ -0,0 +1,204 @@
+    default="results/models/preprocessor/preprocessor.pkl",
+    help="Path to the preprocessor object (default: results/models/preprocessor/preprocessor.pkl)",
+)
+def main(verbose, train, test, output_dir, preprocessor_path):
```
If above comment holds, we should add fit_pickle_to to main parameter list

rbouwer commented on src/english_score_tuning.py:
```diff
@@ -0,0 +1,204 @@
+    best_pipe = make_pipeline(
+        preprocessor, Ridge(alpha=ridge_search.best_params_["ridge__alpha"])
+    )
+    best_pipe.fit(X_train, y_train)
```
@farrandi if you pickle fit best pipe - I will use this in my script

farrandi committed:
change default name from -dataset to -data

atabak-alishiri committed:
fixed the code to work from root

rbouwer commented on src/english_score_tuning.py:
```diff
@@ -0,0 +1,204 @@
+from sklearn.metrics import PredictionErrorDisplay
+from sklearn.model_selection import train_test_split
+from search_top_models import fit_and_return_top_models
+from show_feat_coeff import show_feat_coeff
```
you can remove this now as well as I will use this function

rbouwer replied:
"this" being show_feat_coeff

salva-u committed:
fixed relative path

atabak-alishiri committed:
added default to arguments

farrandi commented:
Tested the pickled model:
```python
# Discussion Section
    if verbose:
        click.echo("Running the Discussion Section...")

    # get best model from pickle file
    with open(ridge_best_model_filename, "rb") as f:
        best_pipe = pickle.load(f)

    # Get the feature coefficient values
    show_feat_coeff(best_pipe, "ridge", preprocessor)

    # plot actual vs predicted
    PredictionErrorDisplay.from_estimator(
        best_pipe, X_test, y_test, kind="actual_vs_predicted", subsample=None
    ).plot()
```

and it works... @rbouwer might be good to reference this code.

farrandi committed:
Saves the best models to pickle file

salva-u committed:
added correlation-matrix csv export

atabak-alishiri committed:
_config.yml file added

farrandi committed:
moved to scripts folder

farrandi committed:
Merge branch 'main' into 60-script-to-tune-hyperparameter-and-analyze

rbouwer commented on src/scripts/english_score_tuning.py:
```diff
@@ -0,0 +1,206 @@
+
+    if verbose:
+        click.echo("Done!")
+
```
hey nando! could you also save the two result dataframes as pngs so that I can show them in the report? I'll put a placeholder for now in the notebook file (in my branch PR) but if you could save them to the figures folder that would be perfect (salva did a similar thing with her correlation matrix for reference)

farrandi replied:
Done!

atabak-alishiri committed:
_toc.yml file added

farrandi committed:
add save to png image

atabak-alishiri committed:
relocated the get_data script

salva-u committed:
finished rerunning download and eda scripts

farrandi committed:
stage results of the tuning script

farrandi commented:
To test the script run: `python src/scripts/english_score_tuning.py -v`
